packages/astro/src/core/config/logging.ts

```diff
@@ -1,13 +1,12 @@
 import type { AstroInlineConfig } from '../../@types/astro.js';
-import type { LogOptions } from '../logger/core.js';
+import { Logger } from '../logger/core.js';
 import { nodeLogDestination } from '../logger/node.js';
 
-export function createNodeLogging(inlineConfig: AstroInlineConfig): LogOptions {
-	// For internal testing, the inline config can pass the raw `logging` object directly
-	if (inlineConfig.logging) return inlineConfig.logging;
+export function createNodeLogger(inlineConfig: AstroInlineConfig): Logger {
+	if (inlineConfig.logging) return new Logger(inlineConfig.logging);
```

Comment:
I think ideally `AstroInlineConfig` has a new `logger` option in place of `logging`, so we could do `if (inlineConfig.logger) return new Logger(inlineConfig.logger);`.

Might have to update tests that passes `logging` to `fixture.startDevServer()` and `fixture.build()` I think. If it turns out to be tricker and we need to keep logging, then I think we should keep the previous comment for now.